app/features-json/project_json_controller.rb

```diff
@@ -25,5 +32,80 @@ class ProjectJSONController < AuthenticatedControllerBase
 
       return project_view_model.to_json
     end
+
+    post HOME do
+      REQUIRED_PARAMS = Set["lane", "repo_org", "repo_name", "project_name", "trigger_type"]
+      has_required_params = REQUIRED_PARAMS.subset?(Set.new(@request_payload.keys))
+
+      if has_required_params
+        provider_credential = check_and_get_provider_credential
+
+        github_service = FastlaneCI::GitHubService.new(provider_credential: provider_credential)
+
+        selected_repo = github_service.repos.detect do |repo|
+          @request_payload["repo_name"] == repo[:name] &&
+            @request_payload["repo_org"] == repo[:owner][:login]
+        end
+
+        repo_config = GitHubRepoConfig.from_octokit_repo!(repo: selected_repo)
```

Comment:
I could imagine the error message is pretty confusing if `selected_repo` is `nil` here, just a guess